database/sqlite/sqlite_functions.c

```diff
@@ -0,0 +1,949 @@
+
+        sqlite3_finalize(chk_size);
+        desired_pages = (sqlite_disk_quota_mb * 0.95) * (1024 * 1024 / page_size);
+        info(
+            "Database desired size is %u pages (page size is %u bytes). Current size is %u pages (includes %u free pages)",
+            desired_pages, page_size, page_count, free_page_count);
+    }
+
+    return ((page_count - free_page_count) / 1024) * (page_size / 1024);
+}
+
+/*
+ * Add a UUID in memory for fast lookup
+ *
+ * Caller should lock the uuid_cache
+ *   Under host holds charts
+ *   Under chart holds dimensions
+ *
+ */
+
+static inline void add_in_uuid_cache(struct uuid_cache **uuid_cache, uuid_t *uuid, const char *type, const char *id, const char *name)
+{
+    //info("add_in_uuid_cache: Adding type=%s, id=%s, name=%s", type?type:"Dimension", id, name?name:"<null>");
+    struct uuid_cache *item = mallocz(sizeof(*item));
+    uuid_copy(item->uuid, *uuid);
+    item->id = id ? strdupz(id) : NULL;
+    item->type = type ? strdupz(type) : NULL;
+    item->name = name ? strdupz(name) : NULL;
+    item->next = *uuid_cache;
+    *uuid_cache = item;
+    return;
+}
+
+/*
+ * Destroy the uuid cache used during startup
+ * Normally it should be empty if all charts and dimensions are active
+ * TODO: Cleanup the cache after the agent starts to save memory
+ * charts / dimensions that will be activated will just query the database
+ */
+void free_uuid_cache(struct uuid_cache **uuid_cache)
+{
+    struct uuid_cache *item;
+    while ((item = *uuid_cache)) {
+        *uuid_cache = item->next;
+        freez(item->id);
+        freez(item->type);
+        freez(item->name);
+        freez(item);
+    }
+    return;
+}
+
+/*
+ * Find a type,id,name in cache -- remove if found
+ * Caller must lock the uuid_cache
+ */
+uuid_t *find_in_uuid_cache(struct uuid_cache **uuid_cache, const char *type, const char *id, const char *name)
+{
+    struct uuid_cache *item;
+    uuid_t *uuid = NULL;
+
+    while ((item = *uuid_cache)) {
+        if ((!strcmp(item->id, id)) && (!type || (!strcmp(item->type, type))) &&
+            ((!name && !item->name) || (name && item->name && !strcmp(item->name, name)))) {
+            uuid = mallocz(sizeof(uuid_t));
+            uuid_copy(*uuid, item->uuid);
+            *uuid_cache = item->next;
+            freez(item->id);
+            freez(item->type);
+            freez(item->name);
+            freez(item);
+            break;
+        }
+        uuid_cache = &(item)->next;
+    }
+    return uuid;
+}
+
+int sql_cache_host_charts(RRDHOST *host)
+{
+#ifdef ENABLE_CACHE_CHARTS
+    int rc;
+    sqlite3_stmt *res = NULL;
+
+    if (!db)
+        return 0;
+
+    if (!res) {
+        rc = sqlite3_prepare_v2(
+            db, "select chart_id, type, id, name from chart where host_id = @host;", -1, &res, 0);
+        if (rc != SQLITE_OK)
+            return 0;
+    }
+
+    rc = sqlite3_bind_blob(res, 1, &host->host_uuid, 16, SQLITE_TRANSIENT);
```

Comment:
Let's try and avoid the magic numbers from now on. We could use something like:
```suggestion
    rc = sqlite3_bind_blob(res, 1, &host->host_uuid, sizeof(host->host_uuid), SQLITE_TRANSIENT);
```
Or use a `#define` if we like it better.